### PR TITLE
Correctly set IsRetargetable for source assemblies

### PR DESCRIPTION
**Pull request by 333fred**
Source assemblies weren't checking assembly flags and setting IsRetargetable, despite having already decoded all well-known attributes to find the other bits of the assembly identity. We now check and correctly set the IsRetargetable bit. Fixes https://github.com/dotnet/roslyn/issues/54836.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs [hunk 1]
```diff
@@ -676,5 +676,17 @@ namespace B;
...
+
+        [Fact, WorkItem(54836, "https://github.com/dotnet/roslyn/issues/54836")]
+        public void AssemblyRetargetableAttributeIsRespected()
+        {
+            var code = @"
+using System.Reflection;
+[assembly: AssemblyFlags(AssemblyNameFlags.Retargetable)]";
+
+            var comp = CreateCompilation(code);
+            Assert.True(comp.Assembly.Identity.IsRetargetable);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs:688]: If I understand correctly, the main purpose of `AssemblyNameFlags.Retargetable` is to cause a flag to be emitted in metadata. 
FWIW, it looks like that was working correctly despite this bug in AssemblyIdentity (see test `AssemblyFlagsAttribute` which checks the emitted metadata).

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by jcouv** [comment]
Looks like there is a legitimate test  failure: `Microsoft.CodeAnalysis.UnitTests.AssemblyIdentityTests.InvalidConstructorArgs`

**Comment by 333fred** [comment]
> Looks like there is a legitimate test  failure: `Microsoft.CodeAnalysis.UnitTests.AssemblyIdentityTests.InvalidConstructorArgs`

Yeah, it's calling the wrong constructor now. I'll need to modify the test in the morning.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

File: src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs:688]: AssemblyFlags is just written out directly, it doesn't read from Identity at all when doing so: https://sourceroslyn.io/#Microsoft.CodeAnalysis/PEWriter/MetadataWriter.cs,1969.

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**